File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.cpp
#include <string>
#include <vector>

using namespace std;

int one_two(int a, int b, int c)
{
    return (a + b + c) * (a*a + b*b + c*c);
}

int solution(int a, int b, int c) {
    int answer = 0;
    
    if (a != b && b != c && c != a)
        answer = a + b + c;
    
    else if (a == b && (b != c || c != a) )
    {
        answer = one_two(a, b, c);
    }
    else if (b == c && (a != b || c != a) )
    {
        answer = one_two(a, b, c);
    }
    else if (c == a && (b != c || a != b) )
    {
        answer = one_two(a, b, c);
    }
    else if (a == b && b == c && c == a)
        answer = (a + b + c) * (a*a + b*b + c*c) * (a*a*a + b*b*b + c*c*c);
        
    return answer;
}

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.py
   
def one_two (a, b, c) :

    return (a + b + c) * (a*a + b*b + c*c)


def solution(a, b, c):
    answer = 0
 
    if a != b and b != c and c != a:
        answer = a + b + c
    
    elif (a == b and (b != c or c != a) ):
        answer = one_two(a, b, c)
    
    elif (b == c and (a != b or c != a) ):
        answer = one_two(a, b, c);
    
    elif (c == a and (b != c or a != b) ):
        answer = one_two(a, b, c)
    
    elif (a == b and b == c and c == a) :
        answer = (a + b + c) * (a*a + b*b + c*c) * (a*a*a + b*b*b + c*c*c)
        
    
    return answer
